### PR TITLE
Remove use of componentWillReceiveProps from date picker

### LINKED ISSUE #836: [Date-Picker] Update React lifecycle methods
# Bug Report

## Description
With the update of react to v16.9.0, `componentWillReceiveProps` lifecycle method has been deprecated. And this outputs a console warning.

## Steps to Reproduce
1. Run jest.

## Additional Context / Screenshots
datepicker

## @ Mentions
@bjankord

### PR DESCRIPTION
**Pull request by pranav300**
### Summary
resolves #836 

### Additional Details
I have taken reference from https://github.com/Hacker0x01/react-datepicker/commit/54434322fe51f52514dd9a464320e604ab66c5cf.

### CODE REVIEW

File: packages/terra-date-picker/src/react-datepicker/index.jsx [hunk 1]
```diff
@@ -389,7 +389,15 @@ class DatePicker extends React.Component {
...
 
-  componentDidUpdate() {
+  componentDidUpdate(prevProps) {
+    const currentMonth = prevProps.selected && getMonth(prevProps.selected)
+    const nextMonth = this.props.selected && getMonth(this.props.selected)
+    if (prevProps.inline && currentMonth !== nextMonth) {
+      this.setPreSelection(this.props.selected)
+    }
+    if (prevProps.highlightDates !== this.props.highlightDates) {
+      this.setState({'highlightDates': getHightLightDaysMap(nextProps.highlightDates)})
```
Comment (saket2403) [inline packages/terra-date-picker/src/react-datepicker/index.jsx]: This should be  this.props.highlightDates

Comment (pranav300) [reply packages/terra-date-picker/src/react-datepicker/index.jsx]: updated here https://github.com/cerner/terra-framework/pull/877/commits/1b2b89d008050997c92f076c9d3738acc8036e2a.

File: packages/terra-date-picker/src/react-datepicker/calendar.jsx [hunk 2]
```diff
@@ -277,14 +277,14 @@ export default class Calendar extends React.Component {
...
-  componentWillReceiveProps (nextProps) {
-    if (nextProps.preSelection && !isSameDay(nextProps.preSelection, this.props.preSelection)) {
+  componentDidUpdate (prevProps) {
+    if (this.props.preSelection && !isSameDay(this.props.preSelection, prevProps.preSelection)) {
       this.setState({
-        date: this.localizeDate(nextProps.preSelection)
+        date: this.localizeDate(this.props.preSelection)
       })
-    } else if (nextProps.openToDate && !isSameDay(nextProps.openToDate, this.props.openToDate)) {
+    } else if (this.props.openToDate && !isSameDay(this.props.openToDate, nextProps.openToDate)) {
```
Comment (saket2403) [inline packages/terra-date-picker/src/react-datepicker/calendar.jsx]: prevProps instead of nextProps

Comment (pranav300) [reply packages/terra-date-picker/src/react-datepicker/calendar.jsx]: Updated here https://github.com/cerner/terra-framework/pull/877/commits/350d1e0ab77f4c65d356bba824a401c61e8dcc53.

**Pull request merged**